cmd/host_logs.go

```diff
@@ -26,24 +27,91 @@ Allowing you to look at the log output generated by the host kernel.
 		section := "host"
 		command := "logs"
 
+		listIdentifiers, _ := cmd.Flags().GetBool("list-identifiers")
+		listBoots, _ := cmd.Flags().GetBool("list-boots")
+
+		if listIdentifiers || listBoots {
+			if listIdentifiers {
+				command += "/identifiers"
+			} else {
+				command += "/boots"
+			}
+
+			resp, err := helper.GenericJSONGet(section, command)
+			if err != nil {
+				fmt.Println(err)
+				ExitWithError = true
+			} else {
+				ExitWithError = !helper.ShowJSONResponse(resp)
+			}
+
+			return
+		}
+
+		identifier, _ := cmd.Flags().GetString("identifier")
+		boot, _ := cmd.Flags().GetString("boot")
+		if len(boot) > 0 {
+			command += "/boots/{boot}"
+		}
+		if len(identifier) > 0 {
+			command += "/identifiers/{identifier}"
+		}
+
+		follow, _ := cmd.Flags().GetBool("follow")
+		if follow {
+			command += "/follow"
+		}
+
 		url, err := helper.URLHelper(section, command)
+
 		if err != nil {
 			fmt.Printf("Error: %v", err)
+			ExitWithError = true
 			return
 		}
 
-		request := helper.GetRequest()
-		resp, err := request.SetHeader("Accept", "text/plain").Get(url)
+		/* Disable timeouts to allow following forever */
+		request := helper.GetRequestTimeout(0).SetHeader("Accept", "text/plain").SetDoNotParseResponse(true)
+
+		lines, _ := cmd.Flags().GetInt32("lines")
+		if lines > 0 {
+			rangeHeader := fmt.Sprintf("entries=:%d:", -(lines - 1))
+			log.WithField("value", rangeHeader).Debug("Range header")
+			request.SetHeader("Range", rangeHeader)
+		}
+
+		if err != nil {
+			fmt.Printf("Error: %v", err)
+			ExitWithError = true
+			return
+		}
+
+		request.SetPathParam("identifier", identifier)
+		request.SetPathParam("boot", boot)
+
+		resp, err := request.Get(url)
 
 		if err != nil {
 			fmt.Println(err)
 			ExitWithError = true
-		} else {
-			fmt.Println(resp.String())
+			return
 		}
+
+		ExitWithError = !helper.StreamTextResponse(resp)
 	},
 }
 
 func init() {
+	hostLogsCmd.Flags().BoolP("follow", "f", false, "Continuously print new log entries")
+	hostLogsCmd.Flags().Int32P("lines", "n", 0, "Number of log entries to show")
+	hostLogsCmd.Flags().StringP("identifier", "t", "", "Show entries with the specified syslog identifier")
+	hostLogsCmd.Flags().StringP("boot", "b", "", "Logs of particular boot ID")
+	hostLogsCmd.Flags().Bool("list-identifiers", false, "Show all syslog identifiers")
+	hostLogsCmd.Flags().Bool("list-boots", false, "Show all boot IDs by offset")
```

Comment:
to be more consistent to our cli maybe we should split that into commands instead flags kinda like
`ha host logs list-boots`